### PR TITLE
Issue 1 initial plugin work

### PR DESCRIPTION
**Pull request by DamnBack**

### CODE REVIEW

File: tests/__init__.py [hunk 1]
```diff
@@ -1,3 +0,0 @@
-# -*- coding: utf-8 -*-
```
Comment (PeterPetrik) [inline tests/__init__.py]: probably to every file add some copyright like:
```
# 3Di API Client for QGIS, licensed under GPLv2 or (at your option) any later version
# Copyright (C) 2020 by Lutra Consulting for 3Di Water Management
```

File: threedi_api_qgis_client/metadata.txt [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+# End of mandatory metadata
+
+# Recommended items:
+
+hasProcessingProvider=no
+# Uncomment the following line and add your changelog:
+# changelog=
+
+# Tags are comma separated with spaces allowed
+tags=python
```
Comment (PeterPetrik) [inline threedi_api_qgis_client/metadata.txt]: would be good to add 3Di, Water Modelling, etc...

File: threedi_api_qgis_client/threedi_api_qgis_client_dockwidget.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    closingPlugin = pyqtSignal()
+
+    def __init__(self, parent=None):
+        """Constructor."""
+        super(ThreediQgisClientDockWidget, self).__init__(parent)
+        # Set up the user interface from Designer.
+        # After setupUI you can access any designer object by doing
+        # self.<objectname>, and you can use autoconnect slots - see
+        # http://doc.qt.io/qt-5/designer-using-a-ui-file.html
+        # #widgets-and-dialogs-with-auto-connect
```
Comment (PeterPetrik) [inline threedi_api_qgis_client/threedi_api_qgis_client_dockwidget.py]: remove

File: threedi_api_qgis_client/threedi_api_qgis_client.py [hunk 4]
```diff
@@ -1,3 +1,199 @@
...
+
+        if os.path.exists(locale_path):
+            self.translator = QTranslator()
+            self.translator.load(locale_path)
+            QCoreApplication.installTranslator(self.translator)
+
+        # Declare instance attributes
+        self.actions = []
+        self.menu = self.tr(u'&3Di API Client')
+        # TODO: We are going to let the user set this up in a future iteration
```
Comment (PeterPetrik) [inline threedi_api_qgis_client/threedi_api_qgis_client.py]: remove

File: threedi_api_qgis_client/threedi_api_qgis_client.py [hunk 5]
```diff
@@ -1,3 +1,199 @@
...
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+"""
+from qgis.PyQt.QtCore import QSettings, QTranslator, QCoreApplication, Qt
+from qgis.PyQt.QtGui import QIcon
+from qgis.PyQt.QtWidgets import QAction
+
+# Import the code for the DockWidget
```
Comment (PeterPetrik) [inline threedi_api_qgis_client/threedi_api_qgis_client.py]: remove

File: tests/__init__.py (hunk 1 above)
Comment (DamnBack) [reply tests/__init__.py]: Done.

File: threedi_api_qgis_client/threedi_api_qgis_client_dockwidget.py (hunk 3 above)
Comment (DamnBack) [reply threedi_api_qgis_client/threedi_api_qgis_client_dockwidget.py]: Done.

File: threedi_api_qgis_client/threedi_api_qgis_client.py (hunk 5 above)
Comment (DamnBack) [reply threedi_api_qgis_client/threedi_api_qgis_client.py]: Done.

File: threedi_api_qgis_client/metadata.txt (hunk 2 above)
Comment (DamnBack) [reply threedi_api_qgis_client/metadata.txt]: Done.

File: threedi_api_qgis_client/threedi_api_qgis_client.py (hunk 4 above)
Comment (DamnBack) [reply threedi_api_qgis_client/threedi_api_qgis_client.py]: Done.

**Pull request merged**